exploration/number-selection.md

```diff
@@ -92,44 +95,274 @@ ICU MF1 messages using `plural` and `selectordinal` should be representable in M
 
 ## Proposed Design
 
-Given that we already have a `:number`,
-it makes sense to add a `<matchSignature>` to it with an option
-
-```xml
-<option name="select" values="plural ordinal exact" default="plural" />
-```
-
-The default `plural` value is presumed to be the most common use case,
-and it affords the least bad fallback when used incorrectly:
-Using "plural" for "exact" still selects exactly matching cases,
-whereas using "exact" for "plural" will not select LDML category matches.
-This might not be noticeable in the source language,
+### Number Selection
+
+Number selection has three modes:
+- `exact` selection matches the operand to explicit numeric keys exactly
+- `plural` selection matches the operand to explicit numeric keys exactly
+  or to plural rule categories if there is no explicit match
+- `ordinal` selection matches the operand to explicit numeric keys exactly
+  or to ordinal rule categories if there is no explicit match
+
+
+### Functions
+
+The following functions use numeric selection:
+
+The function `:number` is the default selector for numeric values.
+
+The function `:plural` operates identically to `:number`, except:
+- the `select` type of this selector is always `plural`
+- the function is **not** valid as a formatter
+
+The function `:ordinal` operates identically to `:number`, except:
+- the `select` type of this selector is always `ordinal`
+  (This function is valid as a formatter)
+
+The function `:integer` operates identically to `:number`, except:
+- the `minimumFractionDigits` and `maximumFractionDigits` options
+  default to `0` for all values
+- the _options_ `minimumFractionDigits` and `maximumFractionDigits`
+  are invalid.
+
+### Operands
+
+The operand of a number selector is either a literal that matches the `number-literal`
+production in the [ABNF](/main/spec/message.abnf) and which is parsed by the 
+implementation into a number; or an implementation-defined numeric type; 
+or is some serialization of a number that the implementation recognizes.
+
+> For example, in Java, any subclass of `java.lang.Number` plus the primitive
+> types (`byte`, `short`, `int`, `long`, `float`, `double`, etc.) 
+> might be considered as the "implementation-defined numeric types".
+> Implementations in other programming languages would define different types
+> or classes according to their local needs.
+
+Implementations MUST treat literals that do not match `number-literal` as
+non-numeric string literals.
```

Comment:
Agreed. Perhaps:

```suggestion
Implementations MUST treat literals that do not match `number-literal` as
plain string literals.
```

Reply:
Your fix doesn't seem to address my issue? The questionable word in this phrase is "literal", because in our common usage it's referring to a value that's defined directly in the syntax source. My concern is that the spec should allow for an implementation to consider the values of e.g. a literal operand `42` and a variable operand with the resolved value of the _string_ `42` to be indistinguishable from each other.

Reply:
That gets into what the "resolved" value is. The *formatted" value of a variable might be `४२` (numerically equal to 42, but not equal in string value).

Reply:
I think my edits this morning (1c0bf19) address this. I removed the MUSTard in favor of text very similar (intentionally similar) to that in date/time. 

@macchiati "resolved value" means mostly that we unwind the chain of assignments to have the actual value in hand. That is:

```
Input: `{ 'a' : 'wildebeest'}`
.local $b = {$a}
.local $c = {$b}
.local $d = {$c}
{{The resolved value of {$d} is the literal 'wildebeest'}}
```

This is not the same thing as the _formatted_ value, although any annotations pick up along the way are available.

FWIW, ` |४२|` is not a valid numeric literal because it does not match the `numeric-literal` production. It's just some string.